### ISSUE TITLE
upgraded to 2.8

### ISSUE THREAD

**Issue by schroef** [Tlousky/production_scripts#5]
I bumped into the addon last night looking for methods of automated node branches. Super nice work! 

Since it quite old and i dont see it updated anymore, is there something else which is hot? I mean i also found the blendernation thread and noticed quite some comments. It is indeed a very handy tool if you use separate layers for compositing. So i was a bit baffled that its not upgraded.

I didnt had to do much work to get it to work in 2.83, at work i have this and at home 2.91. I works now on both versions both onne is Windows the other OSX.

One thing i did change, i wasnt sure how the output worked, didnt really watch the complete video and havent done this kind of rendering using blender. I used an external render engine while back. But im trying to learn more about cycles now.

I wa a bit shocked about that immense node tree being created. So i start to experiment a bit with the positioning.

I think i got something working, looks nice. On one hand it really doesnt matter right, i mean you simply generate and render. If you alter the passes you would generate again. Because all is automated its way less tedious vs manual setup.

PS i did had to change the part which checked for blender version. I guess Blender 2.69 and lower could do separate passes.

Had to quickly learn how to use view_layers. Never really done that neither. I thought i could use objects on view_layer level, but it only works with collections. I tested a simple scene with a floor and 3 objects and hit tons of passes. I still get the result needed. I did make the create folder inactive when single pass is needed. Doesnt make sense and it doesnt do anything if it is selected.

I think the first in this series is the best layout. I prefer this method vs the original because you can check items better

![create-folders-deactivated](https://user-images.githubusercontent.com/6923008/111371484-de0fc780-866f-11eb-9959-496843e293f4.png)

best layout
![new-nodelayout](https://user-images.githubusercontent.com/6923008/111371485-dea85e00-866f-11eb-8a1e-803b7a5c7002.png)

one idea, not so good
![Screen Shot 2021-03-16 at 14 45 47](https://user-images.githubusercontent.com/6923008/111371568-fed81d00-866f-11eb-8a0f-c522ead92cb4.png)

another idea, not so good neither
![Screen Shot 2021-03-16 at 14 47 17](https://user-images.githubusercontent.com/6923008/111371571-00094a00-8670-11eb-97e1-52c300447df5.png)

**Comment by Tlousky** [comment]
@schroef Nice one!
Send me a pull request and I'll integrate your changes so other users will be able to use this script in 2.8+ as well.

**Comment by schroef** [comment]
Ive done some more work. I had an idea of doing a check, i did read about your improvement on poll. Its not completely done. But currently i got it to work when a setup is generated it wil will check if viewlayer names are still the same and if each viewlayer has same items as the generated items. If false it with show exclamation mark.

It was a bit of a hustle to set it up. Im not really at home yet with CollectionProperties and PointerProperties and found Dicts too difficult to run through. So i made a simple list which contain ('view_layer.name','pass.name') i run these against the saved ones and check these in 2 orders. Currently, it seems to work properly

![pass-check-update-1](https://user-images.githubusercontent.com/6923008/111533258-f568b680-873c-11eb-9412-80a67ecc953e.png)
![pass-check-update-2](https://user-images.githubusercontent.com/6923008/111533259-f6014d00-873c-11eb-982b-2cf7e07dbbf8.png)

It would also be nice to do a second check. So when i remove a node setup, it would also show a warning. 

PS i think i need to fork this complete repo just to make a pull on that single file right?

**Comment by schroef** [comment]
I think i got the checker and delete function working nicely. I have not run into issues yet. Was a pain in the ass figure it out. But its super nice now, no more manually deleting it, its not much work ;) 
It also shows if the setup is different vs the viewlayers and passes. It also checks if viewlayers still have the same name.

I only need to check the case with special blend file names. I noticed you added a re.match and pattern. I think ill add Capitalize to that. I first noticed the main folder name was always lower case. But it also had 'scene' hardcoded as a folder name. I adjusted that as well. Most never touch it, but if they do, that now works.


Not sure why by the video wont play?! You need to open it in a new tab, then download it. Chrome doesnt like ?!?
https://user-images.githubusercontent.com/6923008/111588757-5bcff200-879a-11eb-8e30-be96c6d28284.mp4



Still doing some read up on how to be able to do a pull request on a single file. We can do it for a single file, but i still need to fork the complete repo. Not sure i like that.

**Comment by schroef** [comment]
I had another idea, something i use personal. Sometimes when doing a scene with multiple camera i tend to save the camera name in the shot as well. I aslo liked the idea of adding the blendfile name. So i made this custom option, the user can choose between 4 options to add to the file output. If not used it uses the setup you already created.

![srlp-new gui layout](https://user-images.githubusercontent.com/6923008/111856400-bb481200-8900-11eb-9fac-dcb0ad9f0bed.png)

**Comment by vdawg-git** [comment]
Can you update it? This script would be really helpful

**Comment by Tlousky** [comment]
Hi @v-dawg, I unfortunately don't have time to update this script but it appears that others have. You might want to contact @schroef who seemed to have improved it quite a bit.

**Comment by schroef** [comment]
Yeah i believe it got all working. I havent touched it in a while though. Think i was tinkering with adding more items.
Ill check what i got working now and make a repo, most should be working. I remember i did change something about how the nodes where showed show it looked cleaner. I also added different automation as you can see in the work above.

ill try to check this tonight

**Comment by schroef** [comment]
Just had a quick check, seems i need to do more work. Last time i worked on it , was in March, so i get some error now about "Cryptomatte Accurate", Double checking with 2.91 i still have installed, i see a new panel has been added. That means i need to add aditional code to get this working as well. Hope i can do so tonight

![viewlayers-bl291](https://user-images.githubusercontent.com/6923008/143486373-ada438ff-3142-4c30-967a-0b39372072d8.png)
![viewlayers-bl293](https://user-images.githubusercontent.com/6923008/143486377-79b592a5-79f1-4279-a95c-4a6f688d365c.png)

**Comment by schroef** [comment]
Dang had to scratch the back of skull to understand this one again... hahaha welll i got it sort of working again without errors. Though im not sure how i could implement cryptomatte passes in this one. Its kinda complicated how this is done. Normally all outputs have also one output in the render_layer node. However the cryptomatte has 3 for each. I took me while to hack something to gather so it runs through without errors. 

Now i perhaps can tinker how i could possibly add this. I havent really worked with it, saw some videos about. I know its kinda of an awkard setup method.

**Comment by vdawg-git** [comment]
Awesome of you man :)
I hope you can do it :D

**Comment by schroef** [comment]
Caramba, the code now is a bit sloppy :( I found that more items were missing like volume pass and denoise pass. Problem is, these arent directly located under viewlayers but on view_layers.cycles due to this i had to do some hack to get all those passes as well. It seems to work. Yet im not sure how handy that is for post work. I mean all those passes from the denoiser, can folks use these passes in post work?

Was looking at some videos about cryptomatte, not sure how to do this one. Its very specific how its used. I mean i can setup the node and outputs. I could make the setups and then user needs to do some own work. Cryptomatte is a very specific thing

PS im not sure if im making a pull on this repo. I was thinking of making it a separate repo since this repo has more scripts inside which i have not touch nor looked at

**Comment by schroef** [comment]
Just noticed that Cryptomatte also has 2 implementations now in bl293. I see a new node and a legacy node, that legacy is that pain in the ass i was talking about... lol

![cryptomatte-new-vs-legacy](https://user-images.githubusercontent.com/6923008/143518611-3bd4c10d-0337-40f6-9fe3-732a5a03de64.png)

**Comment by schroef** [comment]
Today i just found this addon, it does pehaps do an even better job. THough its outpur is always in folders, it has all the new features already working super nice! Though i find its location in the compositor perhaps weird. You setup the passes before you actually do a render, i would move it to the viewlayer panel. But other then that, it works really flawless. The updating is a bit different. You need to press create nodes each time you change it and it doesnt show something has changed

https://blenderartists.org/t/addon-renderlayer-manager/1250714
